src/app/icd/client/DefaultCheckInDelegate.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "CheckInHandler.h"
+#include <app/icd/client/DefaultCheckInDelegate.h>
+#include <crypto/CHIPCryptoPAL.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+namespace chip {
+namespace app {
+
+CHIP_ERROR DefaultCheckInDelegate::Init(ICDClientStorage * storage)
+{
+    VerifyOrReturnError(storage != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(mpStorage == nullptr, CHIP_ERROR_INCORRECT_STATE);
+    mpStorage = storage;
+    return CHIP_NO_ERROR;
+}
+
+void DefaultCheckInDelegate::OnCheckInComplete(ICDClientInfo & clientInfo)
+{
+    ChipLogProgress(
+        ICD, "Check In Message processing complete: start_counter=%" PRIu32 " offset=%" PRIu32 " nodeid=" ChipLogFormatScopedNodeId,
+        clientInfo.start_icd_counter, clientInfo.offset, ChipLogValueScopedNodeId(clientInfo.peer_node));
+}
+
+void DefaultCheckInDelegate::OnRefreshKey(ByteSpan & keyData)
+{
+    uint8_t randomGeneratedSymmetricKey[chip::Crypto::kAES_CCM128_Key_Length];
+    chip::Crypto::DRBG_get_bytes(randomGeneratedSymmetricKey, sizeof(randomGeneratedSymmetricKey));
+    keyData = ByteSpan(randomGeneratedSymmetricKey);
```

Comment:
I'm not sure this works since the buffer will become invalid after we exit this function.